clang-tools-extra/clangd/unittests/tweaks/ExtractVariableTests.cpp

```diff
@@ -72,6 +72,22 @@ TEST_F(ExtractVariableTest, Test) {
     )cpp";
   EXPECT_UNAVAILABLE(NoCrashCasesC);
 
+  ExtraArgs = {"-xc"};
+  const char *NoCrashDesignator = R"cpp(
+    struct A {
+      struct {
+        int x;
+      };
+    };
+    struct B {
+      int y;
+    };
+    void foo(struct B *b) {
+      struct A a = {.[[x]]=b->y};
```

Comment:
This test case does not fail without the fix.

The reason is that the selection `[[x]]` does not trigger the crash.

The selection needs to be on the `[[->]]` to trigger the crash (and then, the result will be `AVAILABLE`).

Otherwise looks good, thanks!